src/lib.rs

```diff
@@ -110,39 +110,75 @@
 #![deny(warnings)]
 #![cfg_attr(not(feature = "std"), no_std)]
 
-// serde links to std, so go ahead an pull in our own std
-// support in those situations as well.
-#[cfg(feature = "std")]
-extern crate std as core;
-
-#[cfg(feature = "v3")]
-extern crate md5;
-#[cfg(any(feature = "v4"))]
-extern crate rand;
-#[cfg(feature = "v5")]
-extern crate sha1;
-#[cfg(all(feature = "slog", not(test)))]
-extern crate slog;
-#[cfg(all(feature = "slog", test))]
 #[macro_use]
-extern crate slog;
+extern crate cfg_if;
 
-use core::fmt;
-use core::str::FromStr;
+cfg_if! {
+    if #[cfg(feature = "md5")] {
+        extern crate md5;
+    }
+}
 
-#[cfg(feature = "std")]
-mod std_support;
-#[cfg(feature = "serde")]
-mod serde;
+cfg_if! {
+    if #[cfg(feature = "rand")] {
+        extern crate rand;
+    }
+}
 
-#[cfg(feature = "v1")]
-use core::sync::atomic::{AtomicUsize, Ordering};
+cfg_if! {
+    if #[cfg(feature = "serde")] {
+        extern crate serde;
+    }
+}
 
-#[cfg(feature = "v4")]
-use rand::Rng;
+cfg_if! {
+    if #[cfg(feature = "sha1")] {
+        extern crate sha1;
+    }
+}
 
-#[cfg(feature = "v5")]
-use sha1::Sha1;
+cfg_if! {
+    if #[cfg(all(feature = "slog", not(test)))] {
+        extern crate slog;
+    } else if #[cfg(all(feature = "slog", test))] {
+        #[macro_use]
+        extern crate slog;
+    }
+}
+
+cfg_if! {
```

Comment:
Maybe we could reduce some duplicated nesting here by conditionally using `core as std`:

```rust
if #[cfg(not(feature = "std"))] {
    use core as std;
}

use std::fmt;
use std::str;

...
```

What do you think?

Reply:
This PR was intended to switch to `cfg_if`. Another PR should make the that part more readable later... which I think I will have ready soon